### PR TITLE
[Refactor] Set button fullWidth using className

### LINKED ISSUE #176: Refactor Button fullWidth to use className
Button fullWidth is boolean that toggles styles on condition, use conditional className instead and add styles+className at baseStyles without condition.

### PR DESCRIPTION
**Pull request by LJKaski**
## Description

Removed the conditional style application from button baseStyles. Added a  corresponding className to be added conditionally instead.

## Related Issue

Closes #176 

## How Has This Been Tested?

Changes work fine in styleguide locally. Not tested elsewhere yet.

### CODE REVIEW

File: src/core/Button/Button.baseStyles.tsx [hunk 1]
```diff
@@ -121,7 +113,11 @@ export const baseStyles = withSuomifiTheme(
     user-select: none;
   }
 
-  ${fullWidth && fullWidthStyles}
+  &.fi-button--fullWidth {
```
Comment (J-Kallunki) [inline src/core/Button/Button.baseStyles.tsx]: Use lowercase in CSS-classes - with single dash or without

File: src/core/Button/Button.tsx [hunk 2]
```diff
@@ -57,9 +57,13 @@ const StyledButton = styled(
...
       {...passProps}
-      className={classnames(className, {
-        [`${baseClassName}--${variant}`]: variant !== 'default',
-      })}
+      className={classnames(
+        className,
+        {
+          [`${baseClassName}--${variant}`]: variant !== 'default',
+        },
+        fullWidth ? 'fi-button--fullWidth' : '',
```
Comment (J-Kallunki) [inline src/core/Button/Button.tsx]: Uses classnames-function here: https://github.com/JedWatson/classnames
It takes format {'class-name': condition} and can be included using new key in the same object as variant is implemented

File: src/core/Button/Button.baseStyles.tsx (hunk 1 above)
Comment (LJKaski) [reply src/core/Button/Button.baseStyles.tsx]: Fixed.

File: src/core/Button/Button.tsx (hunk 2 above)
Comment (LJKaski) [reply src/core/Button/Button.tsx]: Thanks. Fixed this one as well.

File: src/core/Button/Button.tsx [hunk 3]
```diff
@@ -59,6 +60,7 @@ const StyledButton = styled(
       {...passProps}
       className={classnames(className, {
         [`${baseClassName}--${variant}`]: variant !== 'default',
+        [fullWidthClassName]: fullWidth,
```
Comment (J-Kallunki) [inline src/core/Button/Button.tsx]: This does not need array syntax as `classnames` can read the key-string. Array syntax is needed when using template-literals. Can be seen from the documentation https://github.com/JedWatson/classnames#usage

Comment (J-Kallunki) [reply src/core/Button/Button.tsx]: Actually I was wrong, it needs the Computed property name
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names
as the wanted className is set to the constant not the constant key itself.

**Pull request merged**